index.js

```diff
@@ -0,0 +1,148 @@
+async function action() {
+    const core = require("@actions/core")
+    const exec = require("@actions/exec")
+    const github = require("@actions/github")
+    const octokit = github.getOctokit(process.env.GITHUB_TOKEN);
+
+    const appName = core.getInput(`appName`);
+    if (!appName) {
+        console.log("App Name is required");
+        return; 
+    }
+
+    const action = core.getInput(`action`);
+    if (!action) {
+        console.log("Action is required");
+        return; 
+    }
+    
+    const domainName = core.getInput(`domainName`);
+    const imageTag = core.getInput(`imageTag`);
+    const graffitiSecret = core.getInput(`graffitiSecret`);
+    const accountId = core.getInput(`accountId`);
+
+    if (action === 'install') {
+        
+        if (!domainName) {
+            console.log("domainName is required");
+            return; 
+        }
+
+        if (!imageTag) {
+            console.log("imageTag is required");
+            return; 
+        }
+        
+        if (!graffitiSecret) {
+            console.log("graffitiSecret is required");
+            return; 
+        }
+
+        if (!accountId) {
+            console.log("accountId is required");
+            return; 
+        }
+    }
+    
+    // hostname that will resolve to deployment 
+    const hostname = `www-${github.context.issue.number}.${domainName}`;
+    const namespace = `pr-${github.context.issue.number}`;   
+    // Location of helm chart
+    const chart_path = `charts/${appName}`;
+
+    let cmdOutput = '';
+    let cmdError = '';
+
+    const options = {};
+    options.listeners = {
+        stdout: (data) => {
+            cmdOutput += data.toString();
+        },
+        stderr: (data) => {
+            cmdError += data.toString();
+        }
+    };
```

Comment:
Not massively important (it's not code thats run millions of times) but the following is better for performance:

```suggestion
const options = {
    listeners = {
        stdout: (data) => {
            cmdOutput += data.toString();
        },
        stderr: (data) => {
            cmdError += data.toString();
        }
    }
};
```